packages/site/src/components/PaymentChannelContainer.tsx

```diff
@@ -1,44 +1,89 @@
 import { NitroRpcClient } from "@statechannels/nitro-rpc-client";
-import { PaymentChannelInfo } from "@statechannels/nitro-rpc-client/src/types";
-import { useEffect, useState } from "react";
+import {
+  ChannelStatus,
+  PaymentChannelBalance,
+  PaymentChannelInfo,
+} from "@statechannels/nitro-rpc-client/src/types";
+import { useEffect, useState, FC, useCallback } from "react";
+import { makeStyles } from "tss-react/mui";
 
 import PaymentChannelList from "./PaymentChannelList";
 import PaymentChannelDetails from "./PaymentChannelDetails";
 
-type Props = {
+interface Props {
   nitroClient: NitroRpcClient | null;
   ledgerChannel: string;
+}
+
+const useStyles = makeStyles()(() => ({
+  paymentDetails: {
+    padding: "4rem",
+  },
+}));
+
+const DEFAULT_CHANNEL: PaymentChannelInfo = {
+  ID: "",
+  Status: "Complete" as ChannelStatus,
+  Balance: {
+    AssetAddress: "",
+    Payee: "",
+    Payer: "",
+    PaidSoFar: BigInt(0),
+    RemainingFunds: BigInt(0),
+  } as PaymentChannelBalance,
 };
-export default function PaymentChannelContainer({
+
+const PaymentChannelContainer: FC<Props> = ({
   nitroClient,
   ledgerChannel,
-}: Props) {
+}: Props) => {
   const [paymentChannels, setPaymentChannels] = useState<PaymentChannelInfo[]>(
     []
   );
+  const { classes } = useStyles();
 
   const [focusedPaymentChannel, setFocusedPaymentChannel] =
-    useState<string>("");
+    useState<PaymentChannelInfo>(DEFAULT_CHANNEL);
+
+  const handleGetPaymentChannels = useCallback(
+    async (client: NitroRpcClient, channel: string) => {
+      const paymentChannelsList = await client.GetPaymentChannelsByLedger(
+        channel
+      );
+
+      setPaymentChannels(paymentChannelsList);
+      if (paymentChannelsList.length) {
+        setFocusedPaymentChannel(paymentChannelsList[0]);
+      }
+    },
+    []
+  );
 
   useEffect(() => {
     if (nitroClient && ledgerChannel) {
-      nitroClient.GetPaymentChannelsByLedger(ledgerChannel).then((p) => {
-        setPaymentChannels(p);
-        if (p.length > 0) {
-          setFocusedPaymentChannel(p[0].ID);
-        }
-      });
+      handleGetPaymentChannels(nitroClient, ledgerChannel);
     }
-  }, [nitroClient, ledgerChannel]);
+  }, [nitroClient, ledgerChannel, handleGetPaymentChannels]);
```

Comment:
From my point of view, the code should work, but I use `useCallback` hooks because of next: every time a component **re-renders**, its functions **get recreated**. To prevent this **recreation**, we use the `useCallback` hook to **prevent the function from being recreated unless necessary**. **"Necessary moment"** defines by the hook's props, which we define in square brackets.

The React `useCallback` Hook returns a memoized callback function (think of memoization as caching a value so that it does not need to be recalculated). This allows us to **isolate** resource intensive functions so that **they will not automatically run on every render**.

The `useCallback` Hook only runs when one of its dependencies update.

The usage of `useCallback` hook can improve performance.